ddtrace/ext/ci.py

```diff
@@ -41,28 +43,35 @@
 
 
 def _normalize_ref(name):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_TAGS.sub("", _RE_ORIGIN.sub("", _RE_REFS.sub("", name))) if name is not None else None
 
 
 def _filter_sensitive_info(url):
+    # type: (Optional[str]) -> Optional[str]
     return _RE_URL.sub("\\1", url) if url is not None else None
 
 
 def tags(env=None):
+    # type: (Optional[MutableMapping[str, str]]) -> MutableMapping[str, str]
+    """Extract and set tags from provider environ."""
     env = os.environ if env is None else env
-    tags = {}
+    tags = {}  # type: MutableMapping[str, Optional[str]]
     for key, extract in PROVIDERS:
         if key in env:
             tags = extract(env)
             break
 
+    git_info = git.extract_git_metadata()
+    # Tags collected from CI providers takes precedence over extracted git metadata
+    tags.update({k: v for k, v in git_info.items() if not tags.get(k)})
```

Comment:
```suggestion
    git_info.update(tags)
```
and maybe call `git_info` something else, e.g. `updated_tags`?

Reply:
Made a quick change, let me know what you think 😄